Data/Sys/Shaders/default_pre_post_process.glsl

```diff
@@ -293,6 +293,24 @@ float4 AreaSampling(float3 uvw, float gamma)
 	return avg_color / (area_corners + area_edges + area_center);
 }
 
+/***** 4x Rotated Grid SuperSampling *****/
+
+float4 RGSS(float3 uvw, float gamma)
+{
+	float2 dxy = float2(dFdx(uvw.x), dFdy(uvw.y));
```

Comment:
I don't have experience with these functions, but shouldn't this algorithm account for the input or output resolution?

Reply:
If I have to be honest, I'm a little bit skeptical about the `dFdx` functions. It's mostly because I don't know them, though after giving a look, I'm still scared, for the following reasons:
- It's not in line with the rest of the resampling techniques in this document, and it's harder to undetstand such a complex function,
- These shaders need to compile on DX11, DX12, OpenGL, Metal, Vulkan etc etc. I'm not 100% sure this function would be available or reliable on all?
- It would be nice if these functions were agnotisic to any meta information about how GPUs work (`dFdx` basically returns the ratio of change between the current fragment and its next neighbor).

```
	float2 source_size = GetResolution();
	float2 target_size = GetWindowResolution();
	float2 inverted_target_size = GetInvWindowResolution();
```
I suppose it would still be possible to rely on this data, while having RGSS to give the same exact result?
If not, then I'd bersonally be ok with using `dFdx`, but either way, I don't have a big issue with it.

Reply:
I can use GetInvWindowResolution() if the source [0-1] matches with the window. I dunno what happens with coords when black bars show up and whatnot, my head gets confused.

Reply:
yeah, you don't need to worry about black bars, they are not part of the shader (unless they were drawn by the game in the first place of course). But it's important you understand Dolphin doesn't render with squared pixels, so the image is scaled with a different ratio between X and Y. That doesn't seem to be a problem for this technique.

Reply:
I hope it's okay if I jump in on this even if it's rather late. 

There is a good article on filtering here which is quite relevant to this discussion; https://bgolus.medium.com/sharper-mipmapping-using-shader-based-supersampling-ed7aadb47bec

To summarize, dFdx/dFdy gets the rate of change for a value over a quad (2x2 pixel area). All relevant GPUs render groups of pixels in these 2x2 quads, and this function is available across every API and shading language and should behave the same on each. Between GLSL and HLSL the function has a different name (ddx_fine/ddy_fine vs dFdx and dFdy). And I would say it's a pretty important, basic thing to know about if you want to do antialiasing in a shader. Buuut...

There's also a good article here on the differences in derivatives on various hardware. https://bgolus.medium.com/distinctive-derivative-differences-cce38d36797b 

As this covers a lot of different hardware, I imagine it's possibly relevant to determining whether to use this particular shader trick in Dolphin. And the matter has been settled now that this PR uses the inverse screen resolution instead. But I thought it would be good to provide some additional context from outside.